content/swiftui/concepts/gestures/gestures.md

```diff
@@ -0,0 +1,141 @@
+            }
+        
+    }
+}
+```
+The output looks like:
+
+![Tap_Gesture_SwiftUI.gif](https://raw.githubusercontent.com/Codecademy/docs/main/media/Long_Gesuter_SwiftUI.gif)
+
+ "Long pressed!" was be printed in the console after pressing the text view for 2 seconds.
+
+## Advanced Gestures
+
+For more advanced and customizable gestures use the `gesture()` modifier with one of the gesture structs like: `DragGesture()`, `LongPressGesture()`, `MagnificationGesture()`, `RotationGesture()`, and `TapGesture()`. These all have special modifiers, usually `onEnded()` and `onChanged() `, and they can be used to take action when the gestures are in-flight (for `onChanged()`) or completed (`for onEnded()`).
+
+
+As an example, a magnification gesture could be attached to a view so that pinching in and out scales the view up and down. This can be done by creating two `@State` properties to store the scale amount, using that inside a `scaleEffect()` modifier, then setting those values in the gesture, like this:
+
+```Swift
+struct ContentView: View {
+    @State private var currentAmount = 0.0
+    @State private var finalAmount = 1.0
+
+    var body: some View {
+        Text("Hello, World!")
+            .scaleEffect(finalAmount + currentAmount)
+            .gesture(
+                MagnificationGesture()
+                    .onChanged { amount in
+                        currentAmount = amount - 1
+                    }
+                    .onEnded { amount in
+                        finalAmount += currentAmount 
+                        currentAmount = 0
+                    }
+            )
+    }
+}
+```
+The output looks like:
+
+![Tap_Gesture_SwiftUI.gif](https://raw.githubusercontent.com/Codecademy/docs/main/media/Magnfication_Gesture_SwiftUI.gif)
+
+There is a problem with the code above, because the text view didn't return to it's normal state, this can result in a bad user experience. So to fix this problem the `currentAmount` property will be assigned to zero in the `.onEnded()` modifier, so the text view will return to it's normal size after completing the gesture: 
+
+```Swift
+struct Tap: View {
+    @State private var currentAmount = 0.0
+
+        var body: some View {
+            Text("Hello, World!")
+                .padding()
+                .foregroundColor(.red)
+                .scaleEffect(1 + currentAmount)
+                .gesture(
+                    MagnificationGesture()
+                        .onChanged { amount in
+                            currentAmount = amount - 1
+                        }
+                        .onEnded { amount in
+                            withAnimation(.spring()){
+                                currentAmount = 0
+                            }
+                        }
+                )
+        }
+}
+```
+The output looks like:
+
+![Tap_Gesture_SwiftUI.gif](https://raw.githubusercontent.com/Codecademy/docs/main/media/Magnfication_Gesture_SwiftUI_withAnimation.gif)

```

Comment:
Change the filename on the image, use dashes instead of underscores, and make it all lower case.

Also place some more descriptive text between the `[...]`.